rkt/trust.go

```diff
@@ -0,0 +1,230 @@
+		return "", fmt.Errorf("too many arguments")
+	}
+
+	if prefix == "" {
+		return "", fmt.Errorf("a key or --prefix required")
+	}
+
+	// TODO(vc): discover location from --prefix and return it here?
+
+	return "", nil
+}
+
+// getPubKey retrieves a public key (if remote), and verifies it's a gpg key
+func getPubKey(location string) (*os.File, error) {
+	u, err := url.Parse(location)
+	if err != nil {
+		return nil, err
+	}
+
+	switch u.Scheme {
+	case "":
+		return os.Open(location)
+	case "http":
+		fallthrough
+	case "https":
+		return downloadKey(u.String())
+	default:
+	}
+	return nil, fmt.Errorf("only http and https urls supported")
+}
+
+// downloadKey retrieves the file, storing it in a deleted tempfile
+func downloadKey(url string) (*os.File, error) {
+	tf, err := ioutil.TempFile("", "")
+	if err != nil {
+		return nil, fmt.Errorf("error creating tempfile: %v", err)
+	}
+	os.Remove(tf.Name()) // no need to keep the tempfile around
+
+	res, err := http.Get(url)
+	if err != nil {
+		return nil, fmt.Errorf("error getting key: %v", err)
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("bad HTTP status code: %d", res.StatusCode)
+	}
+
+	if _, err := io.Copy(tf, res.Body); err != nil {
+		tf.Close()
+		return nil, fmt.Errorf("error copying key: %v", err)
+	}
+
+	tf.Seek(0, os.SEEK_SET)
+
+	return tf, nil
+}
+
+func fingerToString(fpr [20]byte) string {
+	str := ""
+	for i, b := range fpr {
+		if i > 0 && i%2 == 0 {
+			str += " "
+			if i == 10 {
+				str += " "
+			}
+		}
+		str += strings.ToUpper(fmt.Sprintf("%.2x", b))
+	}
+	return str
+}
+
+// reviewKey shows the key summary and conditionally asks the user to accept it
+func reviewKey(key *os.File, forceAccept bool) (bool, error) {
+	result := false
+
+	defer key.Seek(0, os.SEEK_SET)
+
+	kr, err := openpgp.ReadArmoredKeyRing(key)
+	if err != nil {
+		return false, fmt.Errorf("error reading key: %v", err)
+	}
+
+	for _, k := range kr {
+		fmt.Printf("GPG key fingerprint is: %s\n", fingerToString(k.PrimaryKey.Fingerprint))
+		for _, sk := range k.Subkeys {
+			fmt.Printf("    Subkey fingerprint: %s\n", fingerToString(sk.PublicKey.Fingerprint))
+		}
+		for n, _ := range k.Identities {
+			fmt.Printf("\t%s\n", n)
+		}
+	}
+
+	if !forceAccept {
+		responded := false
+		for !responded {
+			var input string
+			fmt.Printf("Are you sure you want to trust this key (yes/no)? ")
+			if _, err := fmt.Scanln(&input); err != nil {
+				return result, fmt.Errorf("error reading input: %v", err)
+			}
+			switch input {
+			case "yes":
+				responded = true
+				result = true
+			case "no":
+				responded = true
+			default:
+				fmt.Printf("Please enter 'yes' or 'no'\n")
+			}
+		}
+	}
+	return result, nil
+}
+
+// addPubKey adds a key to the keystore
+func addPubKey(prefix string, key *os.File) error {
+	ks := keystore.New(nil)
+	path, err := ks.StoreTrustedKeyPrefix(prefix, key)
+	if err != nil {
+		return err
+	}
+
+	if globalFlags.Debug {
```

Comment:
I would print this in every case, I don't really see this as a command that is scripted